Apps/Src/UpdateDisplay.c

```diff
@@ -392,4 +391,53 @@ void Task_UpdateDisplay(void *p_arg) {
 			UpdateDisplayError_t err = UpdateDisplay_PopNext();
 			assertUpdateDisplayError(err);
     }
+}
+
+
+/**
+ * Error handler functions
+ * Passed as callback functions to the main assertTaskError function by assertUpdateDisplayError
+*/
+
+/**
+ * @brief A handler callback function run by the main assertTaskError function
+ * used if we haven't reached the restart limit and encounter an error
+ */ 
+static void handler_UpdateDisplay_Restart() {
+	Display_Reset(); // Try resetting to fix the display error
+}
+
+/**
+ * @brief A handler callback function run by the main assertTaskErrorfunction
+ * to show the error screen if we have reached the restart limit
+ */ 
+static void handler_UpdateDisplay_ShowError() {
+	Display_Error(OS_DISPLAY_LOC, Error_UpdateDisplay); // Try resetting the display
+}
+
+
+/**
+ * @brief Check for a display error and assert it if it exists.
+ * Stores the error code, calls the main assertion function 
+ * and runs a callback function as a handler to restart the display or show the error.
+ * @param   err variable with display error codes
+ */static void assertUpdateDisplayError(DisplayError_t err){
```

Comment:
```suggestion
 */
 static void assertUpdateDisplayError(DisplayError_t err){
```

Reply:
This should take in an UpdateDisplayError_t, not a DisplayError_t

Reply:
You're right, thanks